### PR TITLE
Changed case finding report

### PR DESCRIPTION
**Pull request by kkrampa**
@czue @proteusvacuum

### CODE REVIEW

**Comment by czue** [comment]
build fail looks real:
```
======================================================================
ERROR: test_production_config (corehq.apps.userreports.tests.test_static_data_sources.TestStaticDataSource)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vendor/lib/python2.7/site-packages/mock.py", line 1224, in patched
    return func(*args, **keywargs)
  File "/mnt/commcare-hq-ro/corehq/apps/userreports/tests/test_static_data_sources.py", line 52, in test_production_config
    data_source.validate()
  File "/mnt/commcare-hq-ro/corehq/apps/userreports/models.py", line 264, in validate
    raise BadSpecError(_('Report contains duplicate column ids: {}').format(', '.join(set(columns))))
BadSpecError: Report contains duplicate column ids: hiv_positive
----------------------------------------------------------------------
```

**Comment by czue** [comment]
also
```
======================================================================
FAIL: test_extra_pulmonary (custom.enikshay.ucr.tests.test_episode_datasource.TestEpisodeDatasource)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/custom/enikshay/ucr/tests/test_episode_datasource.py", line 136, in test_extra_pulmonary
    self.assertEqual(row.new_smear_positive_extra_pulmonary_TB, 1)
AssertionError: 0 != 1
----------------------------------------------------------------------
```

**Comment by kkrampa** [comment]
@czue @proteusvacuum Fixed :white_check_mark:

File: corehq/apps/userreports/reports/view.py [hunk 1]
```diff
@@ -327,7 +329,7 @@ def headers(self):
     def get_ajax(self, request):
         try:
             data_source = self.data_source
-            if len(data_source.columns) > 50:
+            if not DISABLE_COLUMN_LIMIT_IN_UCR.enabled(self.domain) and len(data_source.columns) > 50:
```
Comment (czue) [inline corehq/apps/userreports/reports/view.py]: super nit picky but you could avoid a DB hit most of the time if you flipped the order of the conditions to check the column limit first

File: corehq/toggles.py [hunk 2]
```diff
@@ -870,3 +870,10 @@ def _commtrackify(domain_name, toggle_is_enabled):
     TAG_PRODUCT_PATH,
     [NAMESPACE_DOMAIN],
 )
+
+DISABLE_COLUMN_LIMIT_IN_UCR = StaticToggle(
+    'disable_column_limit_in_ucr',
+    'Disable column limit in UCR',
+    TAG_EXPERIMENTAL,
```
Comment (czue) [inline corehq/toggles.py]: this should probably be `ONE_OFF` for now

**Comment by czue** [comment]
wow this is a beast. thanks for the updates. i didn't check the report logic too closely but we can update that as part of the QA process (or when @proteusvacuum comes back from vacation if he wants).

**Comment by proteusvacuum** [comment]
Looks good. Would have been good to add some new tests for the indicators that you added, as I think with the `json` format it is pretty hard to parse what is actually going on.
